doc/source/security/encryption-functions.rst

```diff
@@ -0,0 +1,302 @@
+.. _encryption-functions:
+
+===============================================================================
+Encryption functions
+===============================================================================
+
+Percona Server for MySQL 8.0.28-19 added encryption functions. These functions may take an algorithm argument. Encryption converts plaintext into ciphertext using a key and an encryption algorithm. The ciphertext, or encrypted text, can be decrypted by using the same key and algorithm.
+
+A digest uses plaintext and generates a hash value. This hash value can verify if the plaintext is unmodified but you cannot decrypt the original text from the hash value.
+
+When choosing key lengths, consider the following:
+
+* Encryption strength increases the key size and, also, the key generation time.
+
+* If performance is important and the functions are frequently used, use symmetric encryption. Symmetric encryption functions are faster than asymmetric encryption functions.
+
+The following table and sections describe the functions. For examples, see :ref:`function-examples`.
+
+.. list-table::
+   :header-rows: 1
+
+   * - Function Name
+   * - :ref:`asym-decrypt`
+   * - :ref:`asym-derive`
+   * - :ref:`asym-encrypt`
+   * - :ref:`asym-sign`
+   * - :ref:`asym-verify`
+   * - :ref:`create-priv-key`
+   * - :ref:`create-pub-key`
+   * - :ref:`create-dh-param`
+   * - :ref:`create-digest`
+
```

Comment:
Functions never return NULLs.
In exceptional cases, MySQL errors are reported

Reply:
Updated.

Reply:
Removed NULLs